### ISSUE TITLE
Lists

### ISSUE THREAD

**Issue by jmeas** [jmeas/resourceful-redux#175]
This is a proposed solution to #174 

---

tl;dr: A list is an array of resource IDs. You can use this to keep track of "groups" of the same resource.

**But James, we already have those with `labels`!**

You're not wrong. Labels were intended to do double-duty and also act as a list of resources. There are two problems with labels, but keep in mind that you may never run into them:

1. they break the _other_ use case of a label: keeping track of a request. When you delete a resource and it removes the ID from the `label`s array, that's throwing away information about what was returned from the request. This is pretty small, I know, but I think it's preferable if resourceful-redux wouldn't discard information like this.

2. More importantly, labels don't have a good story about supporting reads and creates. Let's say you read a number of resources under the `shoppingCart` label. Great, you have the user's shopping cart. Now the user adds something to their shopping cart. What do you do? You can't add it to the label unless you use the same label when creating the resource. Well, that's not good. I recommend reusing labels, but not across different HTTP methods.

---

With this change, labels serve **one purpose**: they keep track of requests. So you would use a label to determine whether the request has succeeded or failed or not, but you'd use a **list** for determining *what* to display.

---

### Example

First, we dispatch an action to read the shopping cart.

```js
{
  type: actionTypes.READ_RESOURCES,
  resourceName: 'books',
  requestLabel: 'readShoppingCart',
  resourceList: 'shoppingCart',
}
```

Later, we add a book to the shopping cart.

```js
{
  type: actionTypes.CREATE_RESOURCES,
  resourceName: 'books',
  requestLabel: 'addToShoppingCart',
  resourceList: 'shoppingCart',
}
```

In our view layer, we would show the loading indicator when the `readShoppingCart` label is pending. When it comes render time, we would use the `shoppingCart` resource list.

---

### Implementing This

I see this being implemented in 2 phases. The first phase is backwards compatible. Then, there will be a breaking change.

1. a plugin adds support for lists
2. `labels` no longer get updated with DELETEs, `getResources` is updated to no longer accept a label
